third_party/move/move-compiler-v2/tests/checking/typing/bind_unpack_references_invalid.exp

```diff
@@ -1,24 +1,24 @@
 
 Diagnostics:
-error: expected `&u64` but found `integer` in lvalue
+error: `integer` not compatible with `&u64` in lvalue
   ┌─ tests/checking/typing/bind_unpack_references_invalid.move:7:9
```

Comment:
Comparing this with the old (compiler v1) .exp file, I think this would be clearer if we added annotations to the struct declaration/assignment let to state the types which are inferred.  For example:

`     let R { s1: S { f }, s2 } = &R { s1: S{f: 0}, s2: S{f: 1} };`
Contains sub-assignments:
`      f = &0;`
from which we infer f: &uint64.
`     s2 = &S(f:1);`
from which we infer s2 : &M::S.

So this error message would be better as:
error: `integer` not compatible with (inferred) type `&64` of lvalue f.

But if we can't distinguish inferred and declared types, this would still be better as:

error: `integer` type of rvalue not compatible with type `&u64` of lvalue in assignment.

Reply:
Allowing the source code to be made so confusing isn't great for security.

I can imagine someone sharing a module with published source code for clients to scrutinize that looks ok but which does something nefarious with confusing code.